src/NUnitFramework/framework/Internal/Execution/TestWorker.cs

```diff
@@ -66,24 +65,29 @@ public class TestWorker
         /// <param name="apartmentState">The apartment state to use for running tests</param>
         public TestWorker(WorkItemQueue queue, string name, ApartmentState apartmentState)
         {
-            _readyQueue = queue;
-
-            _workerThread = new Thread(new ThreadStart(TestWorkerThreadProc));
-            _workerThread.Name = name;
-            _workerThread.SetApartmentState(apartmentState);
+            WorkQueue = queue;
+            Name = name;
+            Apartment = apartmentState;
```

Comment:
I originally had substantial changes in this file but later found a simpler way to do things. The remaining changes are just cleanup like use of properties... which I left in place. It could be backed out entirely, but it seems to read better this way.